### DISCUSSION TITLE
MyPy compatibility

### DISCUSSION THREAD

**Discussion by graille** [TwitchIO/TwitchIO#362] [Feature requests]
I spent several days to make TwitchIO 2.5 compatible with MyPy. It is for the moment only local and it would need more pronounced tests in order to submit a PR. So I was wondering if there would be any interest in such a PR ?

(I specify that there is no change of code or function signatures, just corrections of type declarations and addition of explicit casts).

**Comment by chillymosh** [comment]
I believe @IAmTomahawkx typed it to Pylance/Pyright instead of MyPy, he will be best to answer this.

**Comment by IAmTomahawkx** [comment]
As chilly said, I've used pyright in 3.0 as it is much more capable of understanding code than mypy. The changes aren't massive to support pyright instead of mypy, but it's a substantially better type checker. Ideally if you could convert this to pyright that would be amazing

**Reply by IAmTomahawkx** [reply]
I should point out that the configurations in https://github.com/TwitchIO/TwitchIO/blob/3.0/pyproject.toml#L56-L83 are ideal settings for us when using pyright
